### PR TITLE
Added new way of testing with Gigaspace in Junit 5

### PR DESCRIPTION
**Pull request by flyhard**
I think using the Old EmbeddedSpace in tests feels very un-junit5-like. I think the new method should work as good while being easier to read.

### CODE REVIEW

**Comment by pativa** [comment]
Can't the existing `EmbeddedSpace` be extended with these new functions?

I don't think we want an extension named `GigaSpaceExtension` that just uses `EmbeddedSpace`. This is a lightweight version of testing small GigaSpaces-related things and I think we want to keep the naming related to that.

I do like the idea of resolving `GigaSpace` as a parameter!

**Comment by flyhard** [comment]
I have now changed it. Take a look :-)

File: README.md [hunk 1]
```diff
@@ -45,17 +45,51 @@ class FruitTest {
...
+	}
+
+	// Test cases against fruitPu
 }
 
 ```
 
+#### Running an Embedded Gigaspace
+
+In case you only need access to gigaspace, you can use the ``EmbeddedSpace`` as a @Rule in Junit 4 or as @ExtendWith for Junit 5:
```
Comment (Bjornelmers) [inline README.md]: Minor: "as a @Rule in Junit 4 or as @ExtendWith for Junit 5" -> "as a @Rule for Junit 4 or as an @ExtendWith for Junit 5"

**Comment by pativa** [comment]
Seems like a good change to improve usability for JUnit5 version of these tests!

**Review by pativa — APPROVED** [review]

**Review by Bjornelmers — APPROVED** [review]
Seems like a good improvement. Nice!

**Pull request merged**